app/src/main/java/com/amaze/filemanager/asynchronous/loaders/AppListLoader.java

```diff
@@ -75,35 +77,39 @@ public AppsDataPair loadInBackground() {
     mApps = new AppsDataPair(new ArrayList<>(apps.size()), new ArrayList<>(apps.size()));
 
     for (ApplicationInfo object : apps) {
-      File sourceDir = new File(object.sourceDir);
-
-      String label = object.loadLabel(packageManager).toString();
-      PackageInfo info;
-
-      try {
-        info = packageManager.getPackageInfo(object.packageName, 0);
-      } catch (PackageManager.NameNotFoundException e) {
-        e.printStackTrace();
-        info = null;
-      }
-
-      AppDataParcelable elem =
-          new AppDataParcelable(
-              label == null ? object.packageName : label,
-              object.sourceDir,
-              object.packageName,
-              object.flags + "_" + (info != null ? info.versionName : ""),
-              Formatter.formatFileSize(getContext(), sourceDir.length()),
-              sourceDir.length(),
-              sourceDir.lastModified(),
-              null);
-
-      mApps.first.add(elem);
-
-      Collections.sort(mApps.first, new AppDataParcelable.AppDataSorter(sortBy, asc));
-
-      for (AppDataParcelable p : mApps.first) {
-        mApps.second.add(p.path);
+      if (object.sourceDir != null) {
```

Comment:
Invert the if to "if(object.sourceDir == null) { <log> continue;}" to reduce indentation level.